helper/cert/source_gen.go

```diff
@@ -217,6 +235,40 @@ func (s *GenSource) getBundleFromSecret() (Bundle, error) {
 	return bundle, nil
 }
 
+func (s *GenSource) getExistingCA(ctx context.Context) []byte {
+	var caBundle []byte
```

Comment:
nit: My personal preference is for less mutability where practical, and I think every case in this switch could just immediately return once it finds a result, rather than needing to track a variable throughout the function.